e3sm_diags/metrics/metrics.py

```diff
@@ -0,0 +1,233 @@
+
+AXES = ["X", "Y"]
+
+
+def get_weights(ds: xr.Dataset):
+    """Get weights for the X and Y spatial axes.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The dataset.
+
+    Returns
+    -------
+    xr.DataArray
+        Weights for the specified axis.
+    """
+    return ds.spatial.get_weights(axis=["X", "Y"])
+
+
+def spatial_avg(
+    ds: xr.Dataset, var_key: str, serialize: bool = False
+) -> xr.DataArray | List[float]:
+    """Compute a variable's weighted spatial average.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The dataset containing the variable.
+    var_key : str
+        The key of the varible.
+    serialize : bool, optional
+        If True, convert the underlying `np.array` to a Python list, by default
+        False. `np.float32` and `np.float64` arrays are not JSON serializable
+        and must be converted to a Python list of native floats if dumping
+        the function's output to a JSON file.
+
+    Returns
+    -------
+    xr.DataArray | List[float]
+        The spatial average of the variable based on the specified axis.
+
+    Raises
+    ------
+    ValueError
+        If the axis argument contains an invalid value.
+
+    Notes
+    -----
+    Replaces `e3sm_diags.metrics.mean`.
+    """
+    ds_avg = ds.spatial.average(var_key, axis=AXES, weights="generate")
+    results = ds_avg[var_key]
+
+    if serialize:
+        return results.data.tolist()
+
+    return results
+
+
+def std(
+    ds: xr.Dataset, var_key: str, serialize: bool = False
+) -> xr.DataArray | List[float]:
+    """Compute the weighted standard deviation for a variable.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The dataset containing the variable.
+    var_key : str
+        The key of the variable.
+    serialize : bool, optional
+        If True, convert the underlying `np.array` to a Python list, by default
+        False. `np.float32` and `np.float64` arrays are not JSON serializable
+        and must be converted to a Python list of native floats if dumping
+        the function's output to a JSON file.
+
+    Returns
+    -------
+    xr.DataArray | List[float]
+        The standard deviation of the variable based on the specified axis.
+
+    Raises
+    ------
+    ValueError
+        If the axis argument contains an invalid value.
+
+    Notes
+    -----
+    Replaces `e3sm_diags.metrics.std`.
+    """
+    dv = ds[var_key].copy()
+
+    weights = ds.spatial.get_weights(axis=AXES, data_var=var_key)
+    dims = _get_dims(dv, axis=AXES)
+
+    result = dv.weighted(weights).std(dim=dims, keep_attrs=True)
+
+    if serialize:
+        return result.data.tolist()
+
+    return result
+
+
+def correlation(
+    da_a: xr.DataArray,
+    da_b: xr.DataArray,
+    weights: xr.DataArray,
+    serialize: bool = False,
+) -> xr.DataArray:
+    """Compute the correlation coefficient between two variables.
+
+    This function uses the Pearson correlation coefficient. Refer to [1]_ for
```

Comment:
Is the underscore supposed to be there? Does that create a footnote in the online docs?

Reply:
Yes, this is a numpy docstring syntax to hyperlink to the "References" link if we ever host API docstrings.

https://github.com/E3SM-Project/e3sm_diags/blob/c02d7015bbd1b211bdf3165d8412ace29ca23d5c/e3sm_diags/metrics/metrics.py#L150-L151